packages/dx-react-grid-material-ui/src/templates/table-filter-cell.jsx

```diff
@@ -4,41 +4,44 @@ import classNames from 'classnames';
 import Input from 'material-ui/Input';
 import { TableCell } from 'material-ui/Table';
 import { withStyles } from 'material-ui/styles';
+import { StaticFilterCellContent } from './static-filter-cell-content';
 
-const styles = theme => ({
-  cell: {
-    verticalAlign: 'top',
-    paddingTop: theme.spacing.unit + 4,
-    paddingRight: theme.spacing.unit,
-    paddingLeft: theme.spacing.unit,
-    '&:first-child': {
-      paddingLeft: theme.spacing.unit * 3,
+const styles = (theme) => {
+  const { spacing } = theme;
+  return {
+    cell: {
+      verticalAlign: 'middle',
+      paddingRight: spacing.unit,
+      paddingLeft: spacing.unit,
+      '&:first-child': {
+        paddingLeft: spacing.unit * 3,
+      },
     },
-  },
-  input: {
-    width: '100%',
-  },
-});
+    input: {
+      width: '100%',
+    },
+  };
+};
 
 const TableFilterCellBase = ({
   style, filter, getMessage, onFilter,
   classes, children, className,
-  tableRow, tableColumn, column,
+  tableRow, tableColumn, column, columnFilteringEnabled,
   ...restProps
 }) => (
   <TableCell
     className={classNames(classes.cell, className)}
     style={style}
     {...restProps}
   >
-    {children || (
+    {columnFilteringEnabled ? (children || (
       <Input
         className={classes.input}
         value={filter ? filter.value : ''}
         placeholder={getMessage('filterPlaceholder')}
         onChange={e => onFilter(e.target.value ? { value: e.target.value } : null)}
       />
-    )}
+    )) : <StaticFilterCellContent>{children}</StaticFilterCellContent>}
```

Comment:
If the `children` property is `undefined` react will generate an error because in the `StaticFilterCellContent` `children` is required. Or I no right?

Reply:
In this case, `children` is marked as required. And it is always defined.